contrib/babelfishpg_tsql/sql/sys_views.sql

```diff
@@ -2594,6 +2594,137 @@ SELECT
 WHERE FALSE;
 GRANT SELECT ON sys.spatial_index_tessellations TO PUBLIC;
 
+
+-- This helper function is used when getting the max length from a value in the typmod array in pg_proc.probin
+CREATE OR REPLACE FUNCTION sys.tsql_typmod_array_max_length_helper(IN type TEXT, IN typelen INT, IN typemod INT, IN for_sys_types boolean DEFAULT false) 
+RETURNS smallint 
+AS $$
+DECLARE
+v_type TEXT COLLATE "C" := type;
+BEGIN
+	
+	IF v_type = 'sysname' THEN
+		RETURN 256;
+	ELSIF (v_type in ('char', 'bpchar', 'varchar', 'binary', 'varbinary', 'nchar', 'nvarchar', 'sysname'))
+	THEN
+		IF typemod < 0 THEN -- max value. 
+			RETURN -1;
+		ELSIF v_type in ('nchar', 'nvarchar') THEN
+			RETURN (2 * typemod);
+		ELSE
+			RETURN typemod;
```

Comment:
It does not match the result of `sys.tsql_type_max_length_helper()` which is `typemod - 4`. Is this intended? If not, I think it would be better to make `sys.tsql_type_max_length_helper()` work in both cases instead of defining a new function.

Reply:
Yes this is intended. The typmod that's passed in here (retrieved from the `typmod_array` in `pg_proc.probin` field) is different than the typmod what should be passed onto `sys.tsql_type_max_length_helper()`.  The typmod value in pg_proc.probin  does not have the added 4. Using the regular `sys.tsql_type_max_length_helper`. 

To add, the one that's passed into `sys.tsql_type_max_length_helper()` should be the typmod value from `pg_attribute` (which would need to be subtracted by 4to get accurate result). Creating a table with `varchar(30)`, will have a typmod value of `34` in `pg_attribute.atttypmod` (which would require a subtraction by 4). But on the otherhand, when creating a proc with a parameter of type `varchar(30)`, the typmod value in the `pg_proc.probin` will be 30.

Reply:
Hi @2jungkook I was able to merge the two by adding another parameter (needed to use the rename method in the upgrade script). But this should avoid having to add an additional helper function